development/gcp/cloud-run/kms-rotate/kms-rotate.go

```diff
@@ -0,0 +1,194 @@
+// package main contains code for KMS GCP symmetric key secret re-encrypting
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"os"
+
+	kms "cloud.google.com/go/kms/apiv1"
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
+)
+
+type Config struct {
+	Project      string `yaml:"project"`
+	Location     string `yaml:"location"`
+	BucketName   string `yaml:"bucketName"`
+	BucketPrefix string `yaml:"bucketPrefix"`
+	Keyring      string `yaml:"keyring"`
+	Key          string `yaml:"key"`
+}
+
+var (
+	kmsService     *kms.KeyManagementClient
+	storageService *storage.Client
+)
+
+func main() {
+	var err error
+	ctx := context.Background()
+
+	kmsService, err = kms.NewKeyManagementClient(ctx)
+	if err != nil {
+		panic("failed creating KMS client, error: " + err.Error())
+	}
+	defer kmsService.Close()
+
+	storageService, err = storage.NewClient(ctx)
+	if err != nil {
+		panic("failed creating KMS client, error: " + err.Error())
+	}
+
+	http.HandleFunc("/", RotateKMSKey)
+	// Determine port for HTTP service.
+	port := os.Getenv("PORT")
+	if port == "" {
+		port = "8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+	// Start HTTP server.
+	log.Printf("Listening on port %s", port)
+	if err := http.ListenAndServe(":"+port, nil); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func RotateKMSKey(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+
+	// get data from POST JSON
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("io.ReadAll: %v", err)
+		http.Error(w, "Couldn't read request body", http.StatusBadRequest)
+		return
+	}
+
+	var conf Config
+
+	json.Unmarshal(body, &conf)
+
+	if conf.Project == "" {
+		log.Print("project value is missing")
+		http.Error(w, "project value is not provided", http.StatusBadRequest)
+		return
+	}
+	if conf.Location == "" {
+		log.Print("location value is missing")
+		http.Error(w, "location value is not provided", http.StatusBadRequest)
+		return
+	}
+	if conf.Keyring == "" {
+		log.Print("keyring value is missing")
+		http.Error(w, "keyring value is not provided", http.StatusBadRequest)
+		return
+	}
+	if conf.Key == "" {
+		log.Print("key value is missing")
+		http.Error(w, "key value is not provided", http.StatusBadRequest)
+		return
+	}
+	if conf.BucketName == "" {
+		log.Print("bucket value is missing")
+		http.Error(w, "bucket value is not provided", http.StatusBadRequest)
+		return
+	}
```

Comment:
You can use a validator of struct fields instead writing conditions for each field. Check if https://github.com/go-playground/validator/v10 suits you. I did use it in https://github.com/kyma-project/test-infra/pull/5544 in file development/tools/pkg/pjtester/pjtester.go line 163.